colorama/ansitowin32.py

```diff
@@ -87,14 +87,21 @@ def __init__(self, wrapped, convert=None, strip=None, autoreset=False):
         # to support the ANSI codes.
         conversion_supported = on_windows and winapi_test()
 
+        # check if this is a TTY or if the FORCE_COLOR environment variable
+        # has been set and NO_COLOR has not been set.
+        isatty = not self.stream.closed and self.stream.isatty()
+        force_color = 'FORCE_COLOR' in os.environ.keys()
+        no_color = 'NO_COLOR' in os.environ.keys()
+        color_allowed = (isatty or force_color) and not no_color
+
         # should we strip ANSI sequences from our output?
         if strip is None:
-            strip = conversion_supported or (not self.stream.closed and not self.stream.isatty())
+            strip = conversion_supported or (not self.stream.closed and not color_allowed)
```

Comment:
Are you sure about this?

In the comment lower down it says "Write the given text to our wrapped stream, stripping any ANSI sequences from the text, and optionally converting them into win32 calls." - so if `NO_COLOR` is specified, for example, we would want colorama to strip the color sequences off any incoming strings.

Or do I misunderstand the meaning?

Reply:
Ah, sorry, I think the name "strip" has been confusing.

If I remember correctly, the name means this:

In normal operation, on Windows, we both "convert" ANSI characters to win32 API calls (thus displaying colors in the terminal), AND then "strip" the ANSI chars from the output (to prevent the ANSI characters from being echoed to the output in the terminal)

So whether a user wants to see colors or not, on Windows we always want to strip the ANSI from the output.